pinc/page_controls.inc

```diff
@@ -0,0 +1,152 @@
+<?php
+include_once($relPath.'stages.inc'); // $Round_for_round_id_
+include_once($relPath.'metarefresh.inc');
+
+function get_page_data()
+{
+    $projectid = array_get($_GET, "project", "");
+    $page = array_get($_GET, "imagefile", "");
+    $round_id = get_enumerated_param($_GET, 'round_id', 'OCR', expanded_rounds());
+    $images = array();
+    $heading = "";
+    $text = "";
+    $error_message = "";
+
+    if($projectid)
+    {
+        try
+        {
+            $projectid = validate_projectID('projectid', $projectid); // throws InvalidProjectIDException
+            $project = new Project($projectid); // throws NonexistentProjectException
+
+            $result = DPDatabase::query("SELECT image FROM {$projectid} ORDER BY image ASC");
+            if(0 == $result->num_rows)
+            {
+                throw new UnexpectedValueException(sprintf(_("There are no pages in %s"), $projectid));
+            }
+            while($row = mysqli_fetch_assoc($result))
+            {
+                $images[] = $row["image"];
+            }
+            mysqli_free_result($result);
+
+            if($page !== "")
+            {
+                // find a match for the page
+                $esc_page = DPDatabase::escape($page);
+                $result = DPDatabase::query("SELECT image FROM $projectid WHERE image LIKE '%$esc_page%'");
```

Comment:
Why are we doing a substring search here? Shouldn't the image name always be one of the ones we've defined? Seems like this would just be:
```
if(!in_array($page, $images))
{
    $page = $images[0];
}
```

Reply:
This was for a request made in a forum https://www.pgdp.net/phpBB3/viewtopic.php?f=4&t=66584&p=1178166&hilit=page+match#p1178168 to use a partial match but now we have all the pages we can do it with php

Reply:
This is going to return unexpected things when multiple pages match. If a project has page "044" and "144" and someone searches for "44" they will get the first entry of whatever the database returns. And since we do not order the select it is up to the database the order that the rows return -- which may or may not be deterministic.

Reply:
I'll write a php function to search in the images we downloaded earlier.